Drop unsampled transactions when connected to APM Server 8.0+ (#554)
elastic/apm: specs/agents/tracing-sampling.md

### Non-sampled transactions

#### Pre 8.0
When connected to an APM Server < 8.0, both sampled and non-sampled transactions MUST be captured by Elastic APM agents.
Sampling controls how much data is captured for transactions:
sampled transactions have complete context recorded and include spans;
non-sampled transactions have limited context and no spans.

For non-sampled transactions set the transaction attributes `sampled: false` and `sample_rate: 0`, and omit `context`.
No spans should be captured.

#### Post 8.0

> When connected to an APM Server 8.0+, agents SHOULD NOT send non-sampled transactions, only transactions that are sampled.
> No spans should be captured for non-sampled transactions.

Reviewer: ```suggestion
When connected to an APM Server 8.0+, agents SHOULD NOT send non-sampled transactions, or capture spans for these transactions.
Sampled transactions MUST be captured by Elastic APM agents.
```
